file.go

```diff
@@ -0,0 +1,35 @@
+package main
+
+import (
+	"fmt"
+	"net/url"
+	"path/filepath"
+	"strings"
+)
+
+type file struct {
+	Name string
+	URL  string
+}
+
+type files []file
+
+func (fs files) String() string {
```

Comment:
you could use a named return value to spare `fileNames := []string{}`
if you do not want to, no need to init the slice, you can just declare it `s := []string` `strings.Join` will work